docs_src/security/Ch-Configuring-Add-On-Services.md

```diff
@@ -1 +1,189 @@
+via `apk add --update` command.
+
+Dockerfile example using alpine-base image and add `dumb-init`:
+
+```dockerfile
+......
+FROM alpine:3.12
+
+# dumb-init needed for injected secure bootstrapping entrypoint script when run in secure mode.
+RUN apk add --update --no-cache dumb-init
+......
+
+```
+
+and then in the service itself should add `/edgex-init/ready_to_run_wait_install.sh` as the entrypoint script
+for the service in gating fashion and related Docker volumes from `edgex-init` and secretstore token which
+will be outlined in the next section.
+
+A good example of this will be like `app-serive-rules`:
+
+```yaml
+...
+  app-service-rules:
+    entrypoint: ["/edgex-init/ready_to_run_wait_install.sh"]
+    command: "/app-service-configurable ${DEFAULT_EDGEX_RUN_CMD_PARMS}"
+    volumes:
+      - edgex-init:/edgex-init:ro,z
+      - /tmp/edgex/secrets/app-rules-engine:/tmp/edgex/secrets/app-rules-engine:ro,z
+    depends_on:
+      - security-bootstrapper
+...
+
+```
+
+## Configure the secret store to use
+
+Make sure the TOML configuration file of add-on service like `device-camera` contains
+the proper `[SecretStore]` section.
+
+Example:
+
+```toml
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = 'my-service/'
+Protocol = 'http'
+RootCaCertPath = ''
+ServerName = ''
+TokenFile = '/tmp/edgex/secrets/my-service/secrets-token.json'
+  [SecretStore.Authentication]
+  AuthType = 'X-Vault-Token'
+```
+
+and then in the EdgeX service `secretstore-setup` environment section of `Docker-compose` file to add
+the service key or hostname to the environment variable `ADD_SECRETSTORE_TOKENS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera'
+...
+
+```
+
+With that, `secretstore-setup` then can generate secretstore token from `Vault` and store it in
+the `TokenFile` path specified in the TOML configuration file like the above example.
+
+## Configure the secure Redis message bus to use
+
+This can be done in the EdgeX service `secretstore-setup` environment section of
+`Docker-compose` file to add the `redisdb` and the service key using it.
+
+So an example will be like `redisdb[device-virtual]` in which `redisdb` is the database resource name
+and the `device-virtual` is the service key or hostname and add that value into the environment variable
+`ADD_KNOWN_SECRETS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera, my-service'
+      ADD_KNOWN_SECRETS: redisdb[app-rules-engine],redisdb[device-rest],redisdb[device-virtual]
+...
+
+```
+
+
+## (Optional) Configure the ACL role of configuration/registry to use if the service depends on it
```

Comment:
Since almost everyone uses Consul, shouldn't this be "Suggested" or "Recommended" instead?

Reply:
No. What I meant is the configuring of ACL role registry is optional if the add-on service doesn't depend on Consul although `Consul` can still be "Suggested" or "Recommended".